docs/proposals/27-cloudprofile-bastion-section.md

```diff
@@ -0,0 +1,141 @@
+Adding the following line should be enough:
+```bash
+systemctl start ssh
+```
+
+## Proposal
+
+We propose to add a new bastion section to the CloudProfile which specifies the bastion creation parameters.
+For the machine image, we want include a reference to one of the `machineImages` in the CloudProfile. We can then use the CloudProfile upon request, to deref that image to the provider specific reference.
+Example:
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: CloudProfile
+metadata:
+  name: cloudprofile1
+spec:
+  type: aws
+  bastion:
+    machineImage:
+      name: gardenlinux
+      version: 1443.3.0
+    machineType:
+      name: n1-standard-2
+...
+```
+
+To be backwards compatible and support CloudProfiles with no bastion section we will use the following fallback strategy:
+
+- Use the first found supported image
+- Use the first found usable machine type
+
+The disadvantage of the fallback strategy is that you implicitly become dependent on the order and in worst case a machine type that is clearly too large could be used for bastions.
+
+## Alternatives
+
+### Place Bastion Configuration Somewhere Else
+
+We can argue if that place should be in the CloudProfile.
+There are other places we could use - like the configuration of the provider-extension.
+The CloudProfile allows us then to have that extra validation to make sure that the image is there.
+Specifying the bastion configuration in the CloudProfile though has the benefit of maintaining everything at one point.
+Otherwise we would need to maintain it for every extension.
+
+### Extending MachineImage and MachineType Section
+
+Instead of explicitly introducing a bastion field to the CloudProfile we could also go for a more general approach.
+What we need mainly need is the `machineImage` and `machineType`.
+We could for example introduce some flag like `default` to the `machineImages` section, which specifies the image family to use:
+```yaml
+...
+machineImages:
+  - name: gardenlinux
+    updateStrategy: minor
+    default: true
+    versions:
+    - architectures:
+      - amd64
+      - arm64
+...
+```
+From this image family we would take the first supported.
+
+Likewise we would add a flag `smallest` for `machineTypes` to identify the machine type to be used when creating the bastion:
+```yaml
+...
+machineTypes:
+  - architecture: amd64
+    cpu: "2"
+    gpu: "0"
+    memory: 4Gi
+    name: g_c2_m4
+    storage:
+      class: standard
+      size: 64Gi
+      type: default
+    usable: true
+    smallest: true
+...
+```
+
+This approach could also be beneficial for other consumers as well.
+For example the Dashboard could prefill fields with those values.
+The disadvantage is that the bastion configuration is not explicit and therefore cannot be directly changed without possible side effects.
```

Comment:
Right. Let's avoid this. "default" and "smallest" is not very clear. I like the explicit configuration that you actually proposed. And generally, using the `CloudProfile` makes sense, since `Bastion`s [reference `Shoot`s](https://github.com/gardener/gardener/blob/0c63ae355302ecf52ae3f38fb0d603f7651b7692/example/100-bastion.yaml#L9-L10), and `Shoot`s reference `CloudProfile`s.